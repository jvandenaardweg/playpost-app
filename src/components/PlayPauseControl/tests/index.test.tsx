import React from 'react';
import { render, RenderAPI } from 'react-native-testing-library';

import { PlayPauseControl } from '../index';

describe('PlayPauseControl', () => {

  describe('rendering', () => {
    let wrapper: RenderAPI;

    beforeAll(() => {
      wrapper = render(<PlayPauseControl isPlaying={false} isLoading={false} onPressPlay={() => {}} />);
    });

    it('should render correctly', () => {
      expect(wrapper.toJSON()).toMatchSnapshot();
    });
  });
});
