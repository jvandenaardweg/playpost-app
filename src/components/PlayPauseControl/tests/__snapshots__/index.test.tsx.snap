// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayPauseControl rendering should render correctly 1`] = `
<View
  accessible={true}
  hitSlop={
    Object {
      "bottom": 20,
      "left": 20,
      "right": 20,
      "top": 20,
    }
  }
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "opacity": 1,
    }
  }
  testID="PlayPauseControl-button"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontSize": 16,
          },
          Object {
            "marginLeft": 3,
          },
          Object {
            "fontFamily": "FontAwesome5Free-Solid",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {
            "fontWeight": "900",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`PlayPauseControlCircle rendering should render correctly 1`] = `
<View
  accessible={true}
  hitSlop={
    Object {
      "bottom": 20,
      "left": 20,
      "right": 20,
      "top": 20,
    }
  }
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "borderRadius": 64,
      "height": 64,
      "justifyContent": "center",
      "opacity": 1,
      "padding": 12,
      "width": 64,
    }
  }
  testID="PlayPauseControlCircle-button"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontSize": 16,
          },
          Object {
            "marginLeft": 3,
          },
          Object {
            "fontFamily": "FontAwesome5Free-Solid",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {
            "fontWeight": "900",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`PlayPauseIcon rendering should render correctly 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": "white",
        "fontSize": 16,
      },
      Object {
        "marginLeft": 3,
      },
      Object {
        "fontFamily": "FontAwesome5Free-Solid",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {
        "fontWeight": "900",
      },
    ]
  }
>
  
</Text>
`;
